### PR TITLE
DiffFilenameChecker to merge all filename based checks

### LINKED ISSUE #32: Complain in PR's when illegal values are found in the diff content
The bot should complain in a PR if the diff contents contain certain unwanted strings like "cfme".

### PR DESCRIPTION
**Pull request by bdunne**

### CODE REVIEW

**Comment by bdunne** [comment]
@Fryguy Please review

**Comment by bdunne** [comment]
Running against https://github.com/ManageIQ/manageiq/pull/7090 returns:

:black_circle: **"cloudforms"** Detected at spec/vcr_cassettes/manageiq/providers/ansible_tower/configuration_manager/refresher.yml:1442
:black_circle: **"cloudforms"** Detected at spec/vcr_cassettes/manageiq/providers/ansible_tower/configuration_manager/refresher.yml:1487
:black_circle: **"cloudforms"** Detected at spec/vcr_cassettes/manageiq/providers/ansible_tower/configuration_manager/refresher.yml:2824

**Comment by bdunne** [comment]
cc @jrafanie

File: app/models/batch_entry.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 class BatchEntry < ActiveRecord::Base
+  serialize :result, Array
```
Comment (Fryguy) [inline app/models/batch_entry.rb]: Can we serialize without the Array bit?  BatchJob and BatchEntry itself is very generic, so making it Array forces it into a certain constraint that isn't really required.

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter.rb [hunk 2]
```diff
@@ -62,17 +58,19 @@ def batch_comment?(comment)
...
-      end
+      message_builder.write("")
+      message_builder.write_lines(content.lines)
       message_builder.comments
     end
 
-    def entries_with_results
-      @entries_with_results ||= batch_job.entries.select(&:result)
+    def merged_results
+      @merged_results = batch_job.entries.collect(&:result).flatten.compact
```
Comment (Fryguy) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter.rb]: You don't need the instance variable.

File: app/workers/commit_monitor_handlers/batch/github_pr_commenter/diff_filename_checker.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+require 'time'
```
Comment (Fryguy) [inline app/workers/commit_monitor_handlers/batch/github_pr_commenter/diff_filename_checker.rb]: Surprised you need this...Rails does this already for you.

**Comment by Fryguy** [comment]
Looks really good...just some minor comments.

**Comment by Fryguy** [comment]
@bdunne I just realized that this broke the enable/disable stuff for particular checkers.  I am ripping out the gemfile labeler, so I'll do it there.

**Pull request merged**